### PR TITLE
Error in updating the form state

### PR DESCRIPTION
**Pull request by Fedor531**
Error in updating the form state

### CODE REVIEW

File: packages/magner/lib/utils/core/mixed-check.ts [hunk 1]
```diff
@@ -53,6 +53,10 @@ export const updateFieldValues = (
   form: Record<string, any>,
   force?: Record<'readOnly' | 'disabled' | 'hidden', boolean | MixedChecker>,
 ) => {
+  // Save pure disabled function
+  if (typeof field.props.disabledCondition !== 'function') {
+    field.props.disabledCondition = field.props.disabled;
+  }
   const disabled = force?.disabled ?? field.props.disabledCondition;
   if (typeof disabled === 'function') {
     field.props.disabled = disabled.bind(null, {
```
Comment (VanishMax) [inline packages/magner/lib/utils/core/mixed-check.ts]: I believe that the last line here `field.props.disabled = disabled.bind` can be changed to `field.props.disabledCondition = field.props.disabled.bind`. In this case, we could remove the saving pure functions code lines. `disabled` prop would stay the same as the user writes it, and only the `disabledCondition` will change.

File: packages/magner/lib/types/form/base.ts [hunk 2]
```diff
@@ -27,17 +27,17 @@ export interface BaseProps {
 
   /** Whether the field is disabled. Accepts static boolean or the dynamic
    * MixedChecker function with access to the user's role and form's state */
-  disabled?: boolean | MixedChecker,
+  disabled?: MixedChecker,
   /** Pure disabled function */
   disabledCondition?: MixedChecker,
```
Comment (VanishMax) [inline packages/magner/lib/types/form/base.ts]: Suggestion: create an interface that would extend the `BaseProps` (with *Condition props), but be used for the inner purposes. In this case, users won't see these properties, but they will exist in the inner context.

For example, imagine it like so:

```ts
export interface BasePropsInner extends BaseProps {
  disabledCondition?: MixedChecker,
  requiredCondition?: MixedChecker,
  readonlyCondition?: MixedChecker,
}
```

And why did you remove the `boolean` flag?

**Review by VanishMax — CHANGES_REQUESTED** [review]

File: packages/magner/lib/utils/core/mixed-check.ts (hunk 1 above)
Comment (Fedor531) [reply packages/magner/lib/utils/core/mixed-check.ts]: @VanishMax Yes, but then you will have to change the logic in the components...

File: packages/magner/lib/types/form/base.ts (hunk 2 above)
Comment (Fedor531) [reply packages/magner/lib/types/form/base.ts]: @VanishMax I removed the boolean flag since the MixedChecker function returns a boolean value anyway.

File: packages/magner/lib/types/form/base.ts [hunk 3]
```diff
@@ -37,6 +43,9 @@ export interface BaseProps {
 
   /** If `readOnly: true`, parse the value of the form field to display as a string */
   readOnlyFormatter?: (val: unknown) => string,
+
+  /** We keep pure functions in this object */
+  inner: InnerConditions
```
Comment (VanishMax) [inline packages/magner/lib/types/form/base.ts]: Suggestion: change the comment to `Magner's inner properties. No need to change`. Right now the comment is somehow misleading

**Review by VanishMax — APPROVED** [review]
